api4/post.go

```diff
@@ -396,6 +397,54 @@ func getPost(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getPostsByIds(c *Context, w http.ResponseWriter, r *http.Request) {
+	postIDs := model.ArrayFromJSON(r.Body)
+
+	if len(postIDs) == 0 {
+		c.SetInvalidParam("post_ids")
+		return
+	}
+
+	postsList, err := c.App.GetPostsByIds(postIDs)
+	if err != nil {
+		c.Err = err
+		return
+	}
+
+	var posts = []*model.Post{}
+
+	for _, post := range postsList {
+
+		channel, err := c.App.GetChannel(post.ChannelId)
```

Comment:
This will call the same channelID repeatedly for every post even if they are from the same channel. We should create a map for this, and make the call only for new channels.

Reply:
I like this idea, added the map